src/expressions/await-expr.md

```diff
@@ -0,0 +1,50 @@
+# Await expressions
+
+> **<sup>Syntax</sup>**\
+> _AwaitExpression_ :\
+> &nbsp;&nbsp; [_Expression_] `.` `await`
+
+Await expressions are legal only within `async` contexts, like an
+`async fn` or an `async` block. They operate on a future. Their effect
+is to "pause" the current computation until the given future is ready
+to produce a value.
+
+More specifically, a `f.await` expression has the following effect.
+
+1. evaluate `f` to a future `tmp` using the `IntoFuture:into_future` method;
+2. `tmp` is pinned using `Pin::new_unchecked`;
+3. This pinned future is then polled by invoking the [`Future::poll`] method and
+   passing it the current [task context] (see below);
+3. if the result returns `Poll::Pending`, then the future returns
+   `Poll::Pending`, suspending its state so that, when re-invoked, we will return
+   to step 2;
+4. if the result returns `Poll::Ready`, then the result is used as the result
+   of the await expression itself.
+
+[_Expression_]: expressions.html
+[`Future::poll`]: https://doc.rust-lang.org/std/future/trait.Future.html#tymethod.poll
+[task context]: https://doc.rust-lang.org/std/task/struct.Context.html
+
+The [task context] refers to the context which was supplied to the
+current async block/fn when it itself was polled. Because `await`
```

Comment:
"it itself" should be changed to "the async context" as well. It took me a minute to figure out what "it" was.